src/auto_renewal.cairo

```diff
@@ -260,21 +294,10 @@ mod AutoRenewal {
             assert(expiry <= block_timestamp + (86400_u64 * 30_u64), 'Domain not set to expire');
 
             // Renew domain
+            // last_renewal is updated before external contract calls to prevent reentrancy attacks
+            // if the naming contract was compromised
             self.last_renewal.write((renewer, root_domain), block_timestamp);
-            let contract = get_contract_address();
-            let erc20 = self.erc20_contract.read();
-            let _tax_contract = self.tax_contract.read();
-            // Transfer limit_price + tax price
-            IERC20CamelDispatcher { contract_address: erc20 }
-                .transferFrom(renewer, contract, limit_price + tax_price);
-            // transfer tax price to tax contract address
-            IERC20CamelDispatcher { contract_address: erc20 }.transfer(_tax_contract, tax_price);
-            // Approve & renew domain
-            IERC20CamelDispatcher { contract_address: erc20 }.approve(naming, limit_price);
-            INamingDispatcher { contract_address: naming }
-                .renew(root_domain, 365_u16, ContractAddressZeroable::zero(), 0, metadata);
-            IERC20CamelDispatcher { contract_address: erc20 }.approve(naming, 0.into());
-
+            // events is sent before calls to other contracts a reordering via reentrancy attack
```

Comment:
I think there is a word missing in this comment,  "to prevent a reordering" no ?

Reply:
exactly